libraries/aws/shadow/src/aws_iot_shadow_api.c

```diff
@@ -285,49 +294,55 @@ static AwsIotShadowError_t _validateDocumentInfo( _shadowOperationType_t type,
             IotLogError( "QoS for Shadow %d must be 0 or 1.",
                          _pAwsIotShadowOperationNames[ type ] );
 
-            IOT_SET_AND_GOTO_CLEANUP( AWS_IOT_SHADOW_BAD_PARAMETER );
+            status = AWS_IOT_SHADOW_BAD_PARAMETER;
         }
     }
 
-    /* Check the retry parameters. */
-    if( pDocumentInfo->retryLimit > 0U )
+    if( status == AWS_IOT_SHADOW_SUCCESS )
```

Comment:
The status checks in this functions seem unnecessary to me since all of the error paths change the status to `AWS_IOT_SHADOW_BAD_PARAMETER`. Is there a problem with just getting rid of the status checks?

Reply:
I will refactor.